File: 21560 - Sai Suraj/week 10/45_Jump Game II.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        jumps = 0
        i=0
        max_reacheable = 0
        last_jump_pos=0
        while(last_jump_pos<len(nums)-1):
            max_reacheable = max(max_reacheable,i+nums[i])
            if(i==last_jump_pos):
                last_jump_pos = max_reacheable
                jumps+=1
            i+=1
        return jumps

File: 21558 - Nithin/week 10/45-jump-game-ii/45-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int currEnd = 0,nextEnd = 0,jumps = 0;

        for(int i = 0;i < nums.size() - 1;i++){
            nextEnd = max(nextEnd, i+nums[i]);
            if(i == currEnd){
                jumps++;
                currEnd = nextEnd;
            }
        }
        return jumps;
    }
};